src/functionaltests/java/com/ericsson/ei/subscriptions/query/FreestyleQueryAggregatedObjectsTestSteps.java

```diff
@@ -0,0 +1,154 @@
+        final String expectedAggrId = "6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43";
+        final String entryPoint = "/query";
+        final String queryAggrObj = "{\"criteria\" :{\"aggregatedObject.id\" : \"" + expectedAggrId + "\"}}";
+
+        LOGGER.debug("Freestyle querying for the AggregatedObject with criteria: " + queryAggrObj);
+        mvcResult = mockMvc.perform(get(entryPoint)
+                .param("request", queryAggrObj)
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+        LOGGER.debug("Response of /query RestApi, Status Code: " + mvcResult.getResponse().getStatus() +
+                           "\nResponse: " + mvcResult.getResponse().getContentAsString());
+        
+        JsonNode jsonNodeResult = objMapper.readValue(mvcResult.getResponse().getContentAsString(), JsonNode.class);
+        JsonNode aggrObjResponse = objMapper.readValue(jsonNodeResult.get(0).get("aggregatedObject").toString(), JsonNode.class); 
+
+        String actualAggrObjId = aggrObjResponse.get("id").asText();
+        LOGGER.debug("AggregatedObject id from Response: " + actualAggrObjId);
+   
+        
+        assertEquals(HttpStatus.OK.toString(), Integer.toString(mvcResult.getResponse().getStatus()));
+        assertEquals("Failed to compare actual Aggregated Object Id:\n" + actualAggrObjId
+                + "\nwith expected Aggregated Object Id:\n" + expectedAggrId,
+                expectedAggrId, actualAggrObjId);
+    }
+    
+    @And("^Perform an invalid freesyle query on Aggregated object$")
+    public void perform_invalid_query_on_newly_created_aggregated_object() throws Throwable {
+        final String invalidAggrId = "6acc3c87-75e0-4b6d-88f5-b1aee4e62b43";
+        final String entryPoint = "/query";
+        final String queryAggrObj = "{\"criteria\" :{\"aggregatedObject.id\" : \"" + invalidAggrId + "\"}}";
+        final String expectedResponse = "[]";
+        
+        LOGGER.debug("Trying an invalid query on /query RestApi with invalid criteria query: " + queryAggrObj);
+        mvcResult = mockMvc.perform(get(entryPoint)
+                .param("request", queryAggrObj)
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+        String responseAsString = mvcResult.getResponse().getContentAsString();
+        int reponseStatusCode = mvcResult.getResponse().getStatus();
+        LOGGER.debug("Response of /query RestApi, Status Code: " + reponseStatusCode +
+                "\nResponse: " + responseAsString);
+    
+        assertEquals(HttpStatus.OK.toString(), Integer.toString(reponseStatusCode));
+        assertEquals("Diffences between actual Aggregated Object:\n" + responseAsString
+                + "\nand expected Aggregated Object:\n" + expectedResponse,
+                expectedResponse, responseAsString);
+    }
+    
+    /*
```

Comment:
Minor comment here but I think Javadoc block is supposed to start with double star like this /**

Reply:
Done